<a href="https://colab.research.google.com/github/HarmonyKM/Project-2/blob/main/Project_2_ML.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project 2**
- **Harmony Gasologa**

### **Import Libraries**

In [1]:
import pandas as pd
import numpy as np

## **Heart Failure**

### **1. Source of Data**

**Heart Failure Prediction Dataset**

[Data Source](https://www.kaggle.com/datasets/fedesoriano/heart-failure-prediction?resource=download)

In [3]:
df_heart = pd.read_csv('/content/drive/MyDrive/heart.csv')
df_heart.head()

Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   
1   49   F           NAP        160          180          0     Normal    156   
2   37   M           ATA        130          283          0         ST     98   
3   48   F           ASY        138          214          0     Normal    108   
4   54   M           NAP        150          195          0     Normal    122   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0  
1              N      1.0     Flat             1  
2              N      0.0       Up             0  
3              Y      1.5     Flat             1  
4              N      0.0       Up             0

### **2. Brief Description of Dataset** 

The Dataset is from Kaggle

The data contains 918 observations with 12 attributes for causing Heart disease.

### **3. Target**

To predict occurence of Heart disease based on 12 attributes or features

### **4. Row Representation**

Row 4 shows:

- Male
- Age 54
- Resting BP : 150

Average Male at age 54 are at High Risk of Heart Failure due to many Factors, though this patient shows a Healthy Heart for his age group

### **5. Classification or Regression Problem**

This is a Classification problem

### **Features and Rows of Dataset**

In [4]:
df_heart.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Age             918 non-null    int64  
 1   Sex             918 non-null    object 
 2   ChestPainType   918 non-null    object 
 3   RestingBP       918 non-null    int64  
 4   Cholesterol     918 non-null    int64  
 5   FastingBS       918 non-null    int64  
 6   RestingECG      918 non-null    object 
 7   MaxHR           918 non-null    int64  
 8   ExerciseAngina  918 non-null    object 
 9   Oldpeak         918 non-null    float64
 10  ST_Slope        918 non-null    object 
 11  HeartDisease    918 non-null    int64  
dtypes: float64(1), int64(6), object(5)
memory usage: 86.2+ KB


- **Has 12 Columns and 917 rows**

### **8. Challenges**

Though the data shows no missing values, the data could be filled with inconsistencies that may be reason for dropped rows or columns that will impact the complete model

## **Spanish Wine**

### **1. Source of Data**

**Spanish Wine Quality Dataset**

[Data Source](https://www.kaggle.com/datasets/fedesoriano/spanish-wine-quality-dataset)

In [5]:
df_wine = pd.read_csv('/content/drive/MyDrive/wines_SPA.csv')
df_wine.head()

winery           wine  year  rating  num_reviews country  \
0  Teso La Monja          Tinto  2013     4.9           58  Espana   
1         Artadi  Vina El Pison  2018     4.9           31  Espana   
2   Vega Sicilia          Unico  2009     4.8         1793  Espana   
3   Vega Sicilia          Unico  1999     4.8         1705  Espana   
4   Vega Sicilia          Unico  1996     4.8         1309  Espana   

             region   price                  type  body  acidity  
0              Toro  995.00              Toro Red   5.0      3.0  
1    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0

### **2. Brief Description of Dataset** 

The Dataset is also from Kaggle

The dataset contains 7500 different types of red wines from Spain with 11 features that describe their price, rating, and even some flavor description. 

They say Red wine is good for the Heart, this relates to my first choice for Project 2

### **3. Target**

The task is to predict either the quality of wine or the prices using the given data.

### **4. Row Representation**

Row 3 shows:

- Winery Vega Sicillia
- Wine Unico
- Year 1996
- An acidity of 3.0

With and acidity that low at the Price of $692.96, is it worth it?

### **5. Classification or Regression Problem**

The datasets can be used for classification or regression tasks. 

### **Features and Rows of Dataset**

In [6]:
df_wine.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7500 entries, 0 to 7499
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   winery       7500 non-null   object 
 1   wine         7500 non-null   object 
 2   year         7498 non-null   object 
 3   rating       7500 non-null   float64
 4   num_reviews  7500 non-null   int64  
 5   country      7500 non-null   object 
 6   region       7500 non-null   object 
 7   price        7500 non-null   float64
 8   type         6955 non-null   object 
 9   body         6331 non-null   float64
 10  acidity      6331 non-null   float64
dtypes: float64(4), int64(1), object(6)
memory usage: 644.7+ KB


- **Has 11 Columns and 7500 rows**

### **8. Challenges**

The dataset is not balanced, and the classes are ordered. 
It has missing many missing values, that usually is a hint for inconsistencies in the dataframe. 